crates/schema-impl/src/key_expansion.rs

```diff
@@ -76,6 +76,9 @@ pub(crate) mod expand_impls {
             //  which converts groups to nested messages.
             encode_key(root_number, field_descriptor_kind.wire_type(), &mut b);
 
+            // TODO remove this assumes the key is string!
```

Comment:
The TODO is a bit cryptic to me. Maybe say something along the lines of: "We are assuming that the key is a string. Therefore, we need to encode the length explicitly to create a valid Protobuf message. Remove the explicit length encoding if we ever change the keys back to be valid Protobuf messages."